airlock_processor/StatusChangedQueueTrigger/__init__.py

```diff
@@ -1,18 +1,141 @@
-    outputEvent.set(
-        func.EventGridOutputEvent(
-            id="step-result-id",
-            data={"tag1": "value1", "tag2": "value2"},
-            subject="test-subject",
-            event_type="test-event-1",
-            event_time=datetime.datetime.utcnow(),
-            data_version="1.0"))
+
+class ContainersCopyMetadata:
+    source_account_name: str
+    source_account_key: str
+    sa_source_connection_string: str
+    sa_dest_connection_string: str
+
+    def __init__(self, source_account_name: str, source_account_key: str, sa_source_connection_string: str, sa_dest_connection_string: str):
+        self.source_account_name = source_account_name
+        self.source_account_key = source_account_key
+        self.sa_source_connection_string = sa_source_connection_string
+        self.sa_dest_connection_string = sa_dest_connection_string
+
+
+def main(msg: func.ServiceBusMessage):
+
+    body = msg.get_body().decode('utf-8')
+    logging.info('Python ServiceBus queue trigger processed message: %s', body)
+
+    try:
+        request_properties = extract_properties(body)
+
+        new_status = request_properties.status
+        req_id = request_properties.request_id
+        ws_id = request_properties.workspace_id
+        request_type = request_properties.type
+    except Exception as e:
+        logging.error(f'Failed processing request - invalid message: {body}, exc: {e}')
+        raise
+
+    logging.info('Processing request with id %s. new status is "%s", type is "%s"', req_id, new_status, type)
+
+    if (is_require_data_copy(new_status)):
+        logging.info('Request with id %s. requires data copy between storage accounts', req_id)
+        containers_metadata = get_source_dest_env_vars(new_status, request_type, ws_id)
+        blob_operations.copy_data(containers_metadata.source_account_name, containers_metadata.source_account_key, containers_metadata.sa_source_connection_string, containers_metadata.sa_dest_connection_string, req_id)
+        return
+
+    # Todo: handle other cases...
+
+
+def extract_properties(body: str) -> RequestProperties:
+    try:
+        json_body = json.loads(body)
+        result = parse_obj_as(RequestProperties, json_body["data"])
+        if not result:
+            raise Exception("Failed parsing request properties")
+    except json.decoder.JSONDecodeError:
+        logging.error(f'Error decoding object: {body}')
+        raise
+    except Exception as e:
+        logging.error(f'Error extracting properties: {e}')
+        raise
+
+    return result
+
+
+def is_require_data_copy(new_status: str):
+    if new_status.lower() in [constants.STAGE_SUBMITTED, constants.STAGE_APPROVED, constants.STAGE_REJECTED, constants.STAGE_BLOCKED]:
+        return True
+    return False
+
+
+def get_source_dest_env_vars(new_status: str, request_type: str, short_workspace_id: str) -> ContainersCopyMetadata:
```

Comment:
IMO the method name isn't telling too much. Given it returns the copy metadata, does it make sense to call it get_copy_operation_metadata or something similar?

Reply:
will fix in my next pr